test/unit/tech/flash.test.js

```diff
@@ -141,3 +145,35 @@ test('seekable', function() {
   result = seekable.call(mockFlash);
   equal(result.length, mockFlash.duration_, 'seekable is empty with a zero duration');
 });
+
+// fake out the <object> interaction but leave all the other logic intact
```

Comment:
So I've actually been moving in the opposite direction, trying to require less code that's unrelated to the test to execute. With the Player and Tech mocks like this one, errors in those classes cause half the tests in the suite to fail. The other way requires more setup but it's also pretty explicit about what the function expects. Though if the class you're mocking changes, I guess you could miss issues around that. So not sure which is better or if there's a middle ground.